curve_apps/driver.py

```diff
@@ -38,42 +39,59 @@ class BaseCurveDriver(BaseDriver):
     _default_name: str
 
     def __init__(self, parameters: BaseData | InputFile):
+        self._out_group = None
         if isinstance(parameters, InputFile):
             parameters = self._parameter_class.build(parameters)
 
         # TODO need to re-type params in base class
         super().__init__(parameters)
 
-    def run(self):
-        """
-        Run method of the driver.
+    @property
+    def out_group(self) -> ContainerGroup | None:
+        """Output container group."""
+
+        if self._out_group is None and self.params.output.out_group is not None:
+            self._out_group = ContainerGroup.create(
+                workspace=self.workspace,
+                name=self.params.output.out_group,
+            )
+
+        return self._out_group
+
+    def store(self, curve: Curve):
         """
-        with self.workspace.open(mode="r+") as workspace:
-            parent = None
-            if self.params.output.out_group is not None:
-                parent = ContainerGroup.create(
-                    workspace=workspace,
-                    name=self.params.output.out_group,
-                )
+        Update container group and monitoring directory.
 
-            name = self.params.output.export_as or self._default_name
+        :param curve: Curve object to store.
+        """
 
-            logger.info("Begin process.")
-            output = self.create_output(name, parent=parent)
-            logger.info("Process completed.")
+        with fetch_active_workspace(self.workspace, mode="r+") as workspace:
+            # with self.workspace.open(mode="r+") as workspace:
+            if self.out_group is not None:
+                curve.parent = self.out_group
```

Comment:
```suggestion
```
Don't think you need this since you are inputting the parent=out_group during creation.

Reply:
correct